Semantics.md

```diff
@@ -112,8 +112,13 @@ or [defined inside the module](Modules.md#linear-memory-section). After import
 or definition, there is no difference when accessing a linear memory whether it
 was imported or defined internally.
 
-In the MVP, linear memory cannot be shared between threads of execution.
-The addition of [threads :unicorn:][future threads] will allow this.
+### Shared Linear Memory
+
+A Linear memory can be marked as shared, which allows it to be shared between
+threads of execution. The shared memory can be imported or defined in the
+module. It is a validation error to attempt to import shared linear
+memory if the module's memory import doesn't specify that it allows shared
+memory.
```

Comment:
There are two places where we can mark a memory as shared:

1. In the `WebAssembly.Memory` constructor. You need to update JS.md for this.
2. In [`memory_type`](https://github.com/WebAssembly/design/blob/master/BinaryEncoding.md#memory_type) (for both import and memory sections). This looks like we don't have room for expansion though?

You also need to update JS.md to specify that `.buffer` will return a `SharedArrayBuffer`.

We also must specify what happens when you `.grow` or use `grow_memory`.